### PR TITLE
[Localization] localize battle stats change message completely

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
1. Add the option to localize message text when battle stats are changed.
2. remove ambiguous 'stats' word when attack, defense, sp.Atk, sp.Def, speed are changed.

## Why am I doing these changes?
1. Currently, pokemon name, stat name, change description message is localized but WHOLE SENTENCE is NOT. (like image)
image
Especially in korean, sometimes postposition(like [[가]]) was incorrect because pre-word and postposition + stat message was separated so that makes Korean users feel weird.

2. When 5 stats probabilistic rank-up(like Ancient Power),
original game message shows all list of stats.
But in pokerogue doesn't show which stats are changed (like image)
image
Because Accuracy, Evasiveness are not included, the word 'stats' seems ambiguous.

## What did change?
- Change the code for localize -> battle stat changing message & Conjunctions ' and '
- Add localization key in each language folders.
- Translate to Korean and move postposition
- Remove ambiguous, hard-coded word 'stats' and make to show stats explicitly which are changed.

### Screenshots/Videos
1. As-is, To-be : English ((1) ~ (3) no changes after localize)

| As-is | To-be |
| :-----: | :---: |
| image | image |
| (1) when single stat is changed | (AGILITY, sharply) |
| image | image |
| (2) when two stats are changed | (CURSE, not ghost type pokemon) |
| image | image |
| (3) when three stats are changed | (QUIVER_DANCE) |
| image | image |
| (4) when five stats are changed | (ANCIENT_POWER) |

3. As-is, To-be : Korean

| As-is | To-be |
| :-----: | :---: |
| image | image |
| (1) when single stat is changed | (AGILITY, sharply) |
| image | image |
| (2) when two stats are changed | (CURSE, not ghost type pokemon) |
| image | image |
| (3) when three stats are changed | (QUIVER_DANCE) |
| image | image |
| (4) when five stats are changed | (ANCIENT_POWER) |

### Manually
In ```src/overrides.ts```
- Set ```MOVESET_OVERRIDE``` to test battle stat change. For example,
```typescript
export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.AGILITY, Moves.CURSE, Moves.QUIVER_DANCE, Moves.ANCIENT_POWER ];
```

## Checklist
- [X] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/zh_CN/battle.ts [hunk 1]
```diff
@@ -62,6 +62,8 @@ export const battle: SimpleTranslationEntries = {
   "drainMessage": "{{pokemonName}} 吸取了体力！",
   "regainHealth": "{{pokemonName}} 回复了体力！",
   "fainted": "{{pokemonNameWithAffix}} 倒下了！",
+  "statChange": "{{pokemonNameWithAffix}}'s {{statsFragment}} {{battleStatLevelChangeDescription}}!",
+  "statsAnd": " and ",
```
Comment (40chyan) [inline src/locales/zh_CN/battle.ts:66]:
```suggestion
  "statChange": "{{pokemonNameWithAffix}}的 {{statsFragment}} {{battleStatLevelChangeDescription}}！",
  "statsAnd": " 和 ",
```

File: src/locales/de/battle.ts [hunk 2]
```diff
@@ -62,6 +62,8 @@ export const battle: SimpleTranslationEntries = {
   "drainMessage": "{{pokemonName}} wurde Energie abgesaugt",
   "regainHealth": "KP von {{pokemonName}} wurden wieder aufgefrischt!",
   "fainted": "{{pokemonNameWithAffix}} wurde besiegt!",
+  "statChange": "{{pokemonNameWithAffix}}'s {{statsFragment}} {{battleStatLevelChangeDescription}}!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:65]:
```suggestion
  "statChange": "{{statsFragment}} von {{pokemonNameWithAffix}} {{battleStatLevelChangeDescription}}!",
```

File: src/locales/de/battle.ts [hunk 3]
```diff
@@ -62,6 +62,8 @@ export const battle: SimpleTranslationEntries = {
...
+  "statsAnd": " and ",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:66]:
```suggestion
  "statsAnd": " und ",
```

**Comment by CodeTappert** [comment]
I would really love to see to have 1 message per stat change instead of the "and"

**Comment by EnochG1** [comment]
@CodeTappert 
Maybe we can do it after this PR approved.
I want to keep the rules of PR - `The PR is self-contained and cannot be split into smaller PRs`
and changing the number of message queue can be split with localize

**Comment by CodeTappert** [comment]
> Reference in new

But it would be smarter to do the split FIRST. And then the localization.

**Comment by EnochG1** [comment]
I think this is the priority. If I split the messages, the game slows down that much.
In my opinion, the first developer has the intention of it.

**Comment by Adri1** [comment]
It have to mention that {{battleStatLevelChangeDescription}} **MUST** have a plural option for "rose" and "fell" fragments.
**Exemple with one stat affected:**

- English: _Pikachu's Attack fell!_
- French: _Attaque de Pikachu baisse !_

**Exemple with TWO stats affected:**

- English: _Pikachu's Attack and Defense fell!_
- French: _Attaque et Défense de Pikachu baisse**nt** !_

And the exact logic of course applies too to "rise" fragments stats.

**Comment by CodeTappert** [comment]
> I have to mention that {{battleStatLevelChangeDescription}} **MUST** have a plural version for all of the "rose" and "fell" fragments. **Exemple with one stat affected:**
> 
> * English: _Pikachu's Attack fell!_
> * French: _Attaque de Pikachu baisse !_
> 
> **Exemple with TWO stats affected:**
> 
> * English: _Pikachu's Attack and Defense fell!_ (nothing changes)
> * French: _Attaque et Défense de Pikachu baisse**nt** !_ (the verb switches to its plural form)
> 
> And the exact logic of course applies too to "rise" fragments stats.
> 
> EDIT: Despite this, I opened a PR anyway to your branch with a translation for French

So even for french it would be better to instead of having 2 stats change in the same sentence to split it into two sentences?

**Comment by EnochG1** [comment]
I would close this request for now.
I found duplicate PR https://github.com/pagefaultgames/pokerogue/pull/2373

**Pull request closed without merging**